appmgr/src/context/rpc.rs

```diff
@@ -301,6 +301,10 @@ impl RpcContext {
     #[instrument(skip(self))]
     pub async fn cleanup(&self) -> Result<(), Error> {
         let mut db = self.db.handle();
+        crate::db::DatabaseModel::new()
+            .package_data()
+            .lock(&mut db, LockType::Write)
+            .await?;
```

Comment:
I compiled everything with `unstable` so there are some changes here related to resolving those issues.